libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -697,6 +697,97 @@ def ImportImageFromStorage(self,
             storage_image_path, image_name))
     return GoogleComputeImage(self.project_id, '', image_name)
 
+  def InsertFirewallRule(self, body: Dict[str, Any]) -> None:
+    """Insert a firewall rule to the project.
+
+    Args:
+      body (Dict): The request body.
+          https://googleapis.github.io/google-api-python-client/docs/dyn/compute_v1.firewalls.html#insert  # pylint: disable=line-too-long
+    """
+
+    logger.info( 'Inserting firewall rule {0:s}, '
+            'targeting tags: {1!s}.'.format(body['name'], body['targetTags'] ))
+    firewall_client = self.GceApi().firewalls()
+    request = firewall_client.insert(project=self.project_id, body=body)
+    response = request.execute()
+    self.BlockOperation(response)
+
+  def AddDenyAllFirewallRules(self,
+                              network: str,
+                              deny_ingress_tag: str,
+                              deny_egress_tag: str,
+                              exempted_src_ips: Optional[List[str]] = None,
+                              enable_logging: bool = False) -> None:
+    """Add deny-all firewall rules, of highest priority.
+
+      Args:
+        network (str): URL of the network resource for thesee firewall rules.
+        deny_ingress_tag (str): Target tag name to apply deny ingress rule.
+        deny_egress_tag (str): Target tag name to apply deny egress rule.
+        exempted_src_ips (List[str]): List of IPs exempted from the deny-all
+          ingress firewall rules, ex: analyst IPs.
+        enable_logging (bool): Optional. Enable firewall logging.
+            Default is False.
+
+      Raises:
+        InvalidNameError: If Tag names are invalid.
+    """
+
+    logger.info('Creating deny-all (ingress/egress) '
+            'firewall rules in {0:s} network.'.format(network))
+
+    if not common.REGEX_DISK_NAME.match(deny_ingress_tag):
```

Comment:
We should probably come up with a better name than `REGEX_DISK_NAME` if we're going to use it on tags too.

Reply:
added a new COMPUTE_RFC1035_REGEX constant to the common.py module and referenced REGEX_DISK_NAME. Left  REGEX_DISK_NAME to not change it everywhere.